collector/file-nr_linux.go

```diff
@@ -0,0 +1,94 @@
+// +build !nonetstat
+
+package collector
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+)
+
+const (
+	procFileNRStat      = "/proc/sys/fs/file-nr"
+	fileNRStatSubsystem = "filenr"
```

Comment:
I think filefd would be a better name, as it's about file descriptors rather than the number of files